atomix/cluster/src/main/java/io/atomix/raft/roles/PassiveRole.java

```diff
@@ -482,24 +480,26 @@ private boolean checkPreviousEntry(
 
     // If the previous log index is less than the last written entry index, look up the entry.
     if (request.prevLogIndex() < lastEntryIndex) {
-      // Reset the reader to the previous log index.
-      reader.seek(request.prevLogIndex());
-
-      // The previous entry should exist in the log if we've gotten this far.
-      if (!reader.hasNext()) {
-        log.debug("Rejected {}: Previous entry does not exist in the local log", request);
-        return failAppend(lastEntryIndex, future);
-      }
+      try (final RaftLogReader reader = raft.getLog().openUncommittedReader()) {
```

Comment:
* Opening a new reader is quite cheap. 
* A new reader is not opened everytime. This case happens only in rare cases. In case you did not read the commit message here is the reasoning for this:
"In this commit, the PassiveRole uses a temporary reader when checking the pre-conditions for accepting an AppendRequest. The reader is only used during special conditions when the previous entry in the log does not match with the expected previous entry. This case happens only rarely. So creating a new reader everytime does not impact performance."

Reply:
Makes sense thanks. I checked the full diff instead of commit by commit, sorry due to that I missed your well descriptive commit messages :see_no_evil: My fault sorry

Reply:
:+1: Ya. I did not mention in the PR description that it is split into the commits. Would you prefer that (for next PRs)? Sometimes I copy commit messages into PR descriptions, but this time I didn't do it :P

Reply:
That it is split in commits is I think not necessary to mention, but maybe that there is useful content in the messages or copying important information yes this would be helpful :)